### PR TITLE
WDPD-33 Credit card configuration

### PR DESCRIPTION
**Pull request by ps-amehic**
### This PR

* adds Credit Card payment method configuration
* shows Credit Card payment method config information on the checkout page
* shows payment method logo on the checkout page
* adds possibility of localising the payment's logo

### How to Test

* Login to the admin and activate the plugin.
* Go to Payment methods and select "Wirecard Credit Card".
* You are able to setup all needed fields for Credit card
* As a customer: You can see in checkout process Credit Card payment method. When selecting, debug info of payment method fields is displayed

### Jira Links
* https://jira.parkside.at/browse/WDPD-33

### CODE REVIEW

File: views/blocks/payment_other.tpl [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<dl>
```
Comment (ps-dfritzsch) [inline views/blocks/payment_other.tpl]: I would call this template file `payment_with_logo` or similar instead as "other" is very generic.

File: metadata.php [hunk 2]
```diff
@@ -77,12 +78,18 @@
...
+            'block' => 'select_payment',
+            'file' => 'views/blocks/paymentselector.tpl'
         )
     ),
     'templates'         => array(
         'transaction.tpl'                   => 'wirecard/paymentgateway/views/admin/tpl/transaction.tpl',
         'transaction_list.tpl'              => 'wirecard/paymentgateway/views/admin/tpl/transaction_list.tpl',
-        'transaction_tab.tpl'               => 'wirecard/paymentgateway/views/admin/tpl/transaction_tab.tpl'
+        'transaction_tab.tpl'               => 'wirecard/paymentgateway/views/admin/tpl/transaction_tab.tpl',
+        'payment_other.tpl'                 => 'wirecard/paymentgateway/views/blocks/payment_other.tpl'
```
Comment (ps-ohanel) [inline metadata.php]: This is not very useful name for the template. Pls use something more describing.

File: Core/OxidEE_Events.php [hunk 3]
```diff
@@ -113,18 +117,38 @@ private static function _extendPaymentMethodTable()
...
+        $sQueryAddThreeDMaid = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_THREE_D_MAID` varchar(128) default '' NOT NULL";
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_THREE_D_MAID', $sQueryAddThreeDMaid);
+
+        $sQueryThreeDSecret = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_THREE_D_SECRET` varchar(128) default '' NOT NULL";
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_THREE_D_SECRET', $sQueryThreeDSecret);
+
+        $sQueryAddMaxLimit = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_NON_THREE_D_MAX_LIMIT` varchar(128) default '' NOT NULL";
```
Comment (ps-dfritzsch) [inline Core/OxidEE_Events.php]: We should use an **integer** representation instead as the 3d limit inputs only allow numbers as far as I can see.

File: Core/OxidEE_Events.php [hunk 4]
```diff
@@ -113,18 +117,38 @@ private static function _extendPaymentMethodTable()
...
+        self::_addColumnIfNotExists(self::PAYMENT_TABLE, 'WDOXIDEE_NON_THREE_D_MAX_LIMIT', $sQueryAddMaxLimit);
+
+        $sQueryAddMinLimit = "ALTER TABLE " . self::PAYMENT_TABLE .
+            " ADD COLUMN `WDOXIDEE_THREE_D_MIN_LIMIT` varchar(128) default '' NOT NULL";
```
Comment (ps-dfritzsch) [inline Core/OxidEE_Events.php]: see: https://github.com/wirecard/oxid-ee/pull/24/files#r270625779

File: views/blocks/payment_other.tpl (hunk 1 above)
Comment (ps-ohanel) [reply views/blocks/payment_other.tpl]: renamed to payment_other_with_logo.tpl to keep up with the Oxid template naming.

File: default_payment_config.xml [hunk 5]
```diff
@@ -7,6 +7,7 @@
         <oxdesc>Wirecard PayPal</oxdesc>
         <oxdesc_1>Wirecard PayPal</oxdesc_1>
         <wdoxidee_logo>paypal.png</wdoxidee_logo>
+        <wdoxidee_logo_1>paypal.png</wdoxidee_logo_1>
```
Comment (ps-cgrach) [inline default_payment_config.xml]: let's discuss this approach

File: views/blocks/payment_other_with_logo.tpl [hunk 6]
```diff
@@ -0,0 +1,55 @@
+[{*
+Uses the content of original OXID payment_other.tpl
+Rendering logo was needed but could not be achieved by extending the block
+*}]
+<dl>
+    <dt>
+        [{assign var="oShopConf" value=$oViewConf->getConfig()}]
+        <input id="payment_[{$sPaymentID}]" type="radio" name="paymentid" value="[{$sPaymentID}]" [{if $oView->getCheckedPaymentId() == $paymentmethod->oxpayments__oxid->value}]checked[{/if}]>
+        [{* Logo added *}]
+        <img src="[{$oShopConf->getShopUrl()}]modules/wirecard/paymentgateway/out/img/[{ $paymentmethod->oxpayments__wdoxidee_logo->value }]">
```
Comment (ps-holler) [inline views/blocks/payment_other_with_logo.tpl]: I'd suggest adding a method to the Payment model for getting the logo URL (e.g. `getLogoUrl()`). That way we don't have any hardcoded paths in the view.

File: views/blocks/payment_other_with_logo.tpl [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                <label class="control-label col-lg-3" for="[{$sPaymentID}]_[{$smarty.foreach.PaymentDynValues.iteration}]">[{$value->name}]</label>
+                <div class="col-lg-9">
+                    <input id="[{$sPaymentID}]_[{$smarty.foreach.PaymentDynValues.iteration}]" type="text" class="form-control textbox" size="20" maxlength="64" name="dynvalue[[{$value->name}]]" value="[{$value->value}]">
+                </div>
+            </div>
+        [{/foreach}]
+
+        <div class="clearfix"></div>
+
+        [{if $paymentmethod->oxpayments__oxid == "wdcreditcard"}]
```
Comment (ps-holler) [inline views/blocks/payment_other_with_logo.tpl]: Is this really supposed to be displayed or is this a debugging leftover?

File: views/blocks/paymentselector.tpl [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[{if substr($paymentmethod->oxpayments__oxid, 0, 2) === "wd"}]
```
Comment (ps-holler) [inline views/blocks/paymentselector.tpl]: We have a helper method for this.

File: Core/OxidEE_Events.php (hunk 3 above)
Comment (ps-ohanel) [reply Core/OxidEE_Events.php]: changed to double

File: Core/OxidEE_Events.php (hunk 4 above)
Comment (ps-ohanel) [reply Core/OxidEE_Events.php]: changed to double

File: views/blocks/paymentselector.tpl (hunk 8 above)
Comment (ps-ohanel) [reply views/blocks/paymentselector.tpl]: Used helper method

File: views/blocks/payment_other_with_logo.tpl (hunk 7 above)
Comment (ps-ohanel) [reply views/blocks/payment_other_with_logo.tpl]: It should be displayed here, as stated in the ticket.

File: default_payment_config.xml (hunk 5 above)
Comment (ps-ohanel) [reply default_payment_config.xml]: Logo localization removed from this PR

File: views/blocks/payment_other_with_logo.tpl (hunk 6 above)
Comment (ps-ohanel) [reply views/blocks/payment_other_with_logo.tpl]: Logo code removed from this PR

**Pull request closed without merging**